applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp

```diff
@@ -290,24 +294,53 @@ namespace MPMParticleGeneratorUtility
                         // Get shape_function_values from defined particle_per_condition
                         const Geometry< Node >& r_geometry = i->GetGeometry(); // current condition's geometry
 
-                        Matrix shape_functions_values;
+                        // Check number of particles per condition to be created
+                        bool is_equal_int_volumes = false; // default GAUSS
+                        if (i->Has( IS_EQUAL_DISTRIBUTED )){
+                            is_equal_int_volumes = i->GetValue(IS_EQUAL_DISTRIBUTED);
+                        }
+
+                        std::vector<IntegrationPoint<3>> integration_points;
+                        IndexType number_of_points_per_span;
+                        array_1d<double, 3> local_coordinates;
+                        array_1d<double, 3> xg;
```

Comment:
Why are you defining the variable `xg` instead of using `mpc_xg` (already defined at the beginning of the method)?

Reply:
you are right - I can remove this additional variable